src/models/tag/stream.rs

```diff
@@ -24,6 +23,78 @@ pub enum TagStreamReach {
     Friends,
 }
 
+#[derive(Deserialize, Debug, ToSchema, Clone)]
+pub enum TaggedType {
+    Post,
+    User,
+}
+
+impl Display for TaggedType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TaggedType::Post => write!(f, "Post"),
+            TaggedType::User => write!(f, "User"),
+        }
+    }
+}
+
+#[derive(Deserialize, Debug, ToSchema, Clone)]
+pub enum Timeframe {
+    Today,
+    ThisMonth,
+    AllTime,
+}
+
+impl Display for Timeframe {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Timeframe::Today => write!(f, "Today"),
+            Timeframe::ThisMonth => write!(f, "ThisMonth"),
+            Timeframe::AllTime => write!(f, "AllTime"),
+        }
+    }
+}
+
+impl Timeframe {
+    pub fn to_timestamp_range(&self) -> (i64, i64) {
+        let now = chrono::Utc::now();
+        let start = match self {
+            Timeframe::Today => now
+                .date_naive()
+                .and_hms_opt(0, 0, 0)
+                .unwrap()
+                .and_utc()
+                .timestamp_millis(),
+            Timeframe::ThisMonth => now
+                .date_naive()
+                .with_day(1)
+                .unwrap()
+                .and_hms_opt(0, 0, 0)
+                .unwrap()
+                .and_utc()
+                .timestamp_millis(),
+            Timeframe::AllTime => 0,
+        };
+        (start, now.timestamp_millis())
+    }
+
+    pub fn to_cache_period(&self) -> i64 {
+        match self {
+            Timeframe::Today => 60 * 60,
+            Timeframe::ThisMonth => 60 * 60 * 24,
+            Timeframe::AllTime => 60 * 60 * 24,
```

Comment:
Looks like we are re-computing and caching `?timeframe=all_time&reach=all` every 24 hours. While these are already always available and always up-to-date directly from Index, is this correct @tipogi ? We might need a bit of triaging `can_use_index()` as in `PostStream` to always retrieve from the most efficient DB.